js/shopping-list.js

```diff
@@ -0,0 +1,50 @@
+var ulElement = document.querySelector('ul');
+
+function addListItems(event){
+    var form = event.target;
+    var items = form.querySelector('#items').value.split(',');
+
+    if(items[0].length > 0){
+        document.querySelector('#user-warning').textContent = "";
+        for(var i = 0; (i < items.length) && (items[i].length > 0) && (items[i] !== ' '); i++){
+            appendItemToList(createXImgElement(), items[i]);
+        }
+    } else {
+        document.querySelector('#user-warning').textContent = "Please enter an item.";
+    }
+    event.preventDefault();
+    form.reset();
+}
+
+function createXImgElement(){
+    var imgElement = document.createElement('img');
+    imgElement.setAttribute('src', 'media/x-icon.png');
+    imgElement.setAttribute('alt', 'A red x to indicate removal.');
+    imgElement.addEventListener('click', removeListItem);
+    return imgElement;
+}
+
+function appendItemToList(imgElement, item){
+    var liElement = document.createElement('li');
+    liElement.textContent = item;
+    liElement.appendChild(imgElement);
+    ulElement.appendChild(liElement);
+}
+
+function removeListItem(event){
+    var imgElement = event.target;
+    var liElement = imgElement.parentNode;
+    ulElement.removeChild(liElement);
+}
+
+function clearList(){
+    while(ulElement.hasChildNodes()){
```

Comment:
Interesting find - I've been using `while (element.firstChild)` and relying on truthy values, but something that explicitly returns a Boolean is nice. I may look at swapping out logic at work with this method instead.